### PR TITLE
feat(gatsby-starter-minimal): Add 404 to minimal starter

### PR DESCRIPTION
**Pull request by ascorbic**
Adds a default 404 page to the minimal starter. It's based on the default index and the gatsbyjs.com 404. It includes an extra sentence of help if running in development.

image

### CODE REVIEW

File: starters/gatsby-starter-minimal/src/pages/404.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  return (
+    <main style={pageStyles}>
+      <title>Not found</title>
+      <h1 style={headingStyles}>Page not found</h1>
+      <p style={paragraphStyles}>
+        Sorry{" "}
+        <span role="img" aria-label="Pensive emoji">
+          😔
+        </span>{" "}
+        we couldn’t find what you were looking for. <Link to="/">Go home</Link>.
```
Comment (fk) [inline starters/gatsby-starter-minimal/src/pages/404.js:38]:
```suggestion
        we couldn’t find what you were looking for.<br />
        <Link to="/">Go home</Link>.
```

🤔

**Review by mxstbr — DISMISSED** [review]
LGTM, makes a lot of sense! 👍

File: starters/gatsby-starter-minimal/src/pages/404.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      </p>
+      {process.env.NODE_ENV === "development" ? (
+        <>
+          <p>
+            Try creating a page in <code style={codeStyles}>src/pages/</code>.
+          </p>
+          <ul style={listStyles}>
+            {links.map(link => (
+              <li style={listItemStyles}>
```
Comment (fk) [inline starters/gatsby-starter-minimal/src/pages/404.js:47]: The `You can edit this page at ` from https://github.com/gatsbyjs/gatsby/pull/28636#issue-540313311 is rendered in an  `<li>`?

**Review by fk — APPROVED** [review]
🙏 :shipit: Left two q's, maybe we can chat abt this a bit tomorrow.

**Comment by laurieontech** [comment]
Looks like there was an errant merge with the index page listing links, so removed that. And adjusted ordering/styling so these are the two 404 pages.

Screen Shot 2020-12-15 at 1 42 09 PM
Screen Shot 2020-12-15 at 1 42 45 PM

**Review by fk — APPROVED** [review]
<3 🙏 🚢 like it's hot

**Pull request merged**